src/lib/marshal/shared/helpers.ts

```diff
@@ -0,0 +1,81 @@
+import * as path from "node:path";
+
+import * as fs from "fs-extra";
+
+import { formatErrors, JsonDataError } from "@/lib/helpers/error";
+import { ParsedJson, parseJson } from "@/lib/helpers/json";
+import { validateLiquidSyntax } from "@/lib/helpers/liquid";
+import { LayoutDirContext, WorkflowDirContext } from "@/lib/run-context";
+
+import { VISUAL_BLOCKS_JSON } from "../workflow";
+
```

Comment:
So this is the extracted files read logic from the `worklfow reader`. Now that we use this for layouts as well, I think it makes sense to move this to a shared file.

The location and name of the folder can be other if you disagree with this one!